### ISSUE TITLE
auto-save files

### ISSUE THREAD

**Issue by herrsimon** [mbrlabs/Lorien#174]
First of all, thanks for this great piece of software! I'm a lecturer and intend to test drive it in some of my classes. It would be nice  to add an auto-save feature with configurable time interval. The autosave directory could be `$XDG_DATA_HOME/lorien/autosaves`, the autosave files themselves could be `path_to_a_file` if the original file is `/path/to/a/file` or a simple timestamped generic name in case the canvas has never been saved before.

**Comment by flo-schu** [comment]
I second that. I'm really enjoying the software so far, but just lost an hour of sketching, because lorien crashed unexpectedly.

**Comment by sleeplessKomodo** [comment]
Agreed. This is currently my favourite application for mind maps and free-form note-taking. I just lost about an hour's worth of notes, but of course, I feel silly because I usually would have saved, just not in this instance.  

I should note that it crashed due to [this bug (Uneraseable strokes #261)](https://github.com/mbrlabs/Lorien/issues/261), as I was attempting to erase a stroke, the application crashed, and I didn't think to tee the application's output to a log file.

**Comment by fraburnham** [comment]
I took a stab at autosave, though I went a slightly different way than @herrsimon suggested. I targeted v0.6.0 since that's the version I had installed. If the implementation looks good I can make it work with godot 4 so it can merge into v0.7.0 @mbrlabs.

https://github.com/fraburnham/Lorien/pull/1

This is my first time using godot so please let me know if anything could be done better! I'm happy to make changes!
